### PR TITLE
fix: clicking the header cell does not always sets the focus

### PR DESCRIPTION
**Pull request by LiKang6688**
- When clicking TableSortLabel, it gets focused instead of the header cell.
So adding an explicit click method to TableSortLabel adds focus to the header cell.

- Add a cursor to the head cell

### CODE REVIEW

File: src/table/components/TableHeadWrapper.jsx [hunk 1]
```diff
@@ -42,6 +42,11 @@ function TableHeadWrapper({
               setFocusedCellCoord,
             });
           };
+          const handleMouseDown = (evt) => {
```
Comment (haxxmaxx) [inline src/table/components/TableHeadWrapper.jsx:45]: their are multiple mouse downs in this file, this name is too generic.

File: src/table/components/TableHeadWrapper.jsx [hunk 2]
```diff
@@ -42,6 +42,11 @@ function TableHeadWrapper({
...
+            evt.preventDefault();
+            const cell = getCellElement(rootElement, [columnIndex, 0]);
+            updateFocus({ focusType: 'focus', cell });
```
Comment (haxxmaxx) [inline src/table/components/TableHeadWrapper.jsx:48]: what happens if all this function does is preventDefault? Does the event bubble to onMouseDown on the cell?

Comment (LiKang6688) [reply src/table/components/TableHeadWrapper.jsx:48]: Yes, the event bubbles further up the chain as evt.stoppropagation() is intentionally not used.

https://developer.mozilla.org/en-US/docs/Learn/JavaScript/Building_blocks/Events#fixing_the_problem_with_stoppropagation

Comment (haxxmaxx) [reply src/table/components/TableHeadWrapper.jsx:48]: so then do you need to to anything else here than preventDefault?

File: src/table/utils/handle-accessibility.js [hunk 3]
```diff
@@ -1,5 +1,5 @@
 export const getCellElement = (rootElement, cellCoord) =>
-  rootElement.getElementsByClassName('sn-table-row')[cellCoord[0]]?.getElementsByClassName('sn-table-cell')[
+  rootElement?.getElementsByClassName('sn-table-row')[cellCoord[0]]?.getElementsByClassName('sn-table-cell')[
```
Comment (haxxmaxx) [inline src/table/utils/handle-accessibility.js:2]: how can rootElement be undefined?

File: src/table/components/TableHeadWrapper.jsx (hunk 1 above)
Comment (LiKang6688) [reply src/table/components/TableHeadWrapper.jsx]: Fixed

File: src/table/utils/handle-accessibility.js (hunk 3 above)
Comment (LiKang6688) [reply src/table/utils/handle-accessibility.js]: Fixed

File: src/table/components/TableHeadWrapper.jsx (hunk 2 above)
Comment (LiKang6688) [reply src/table/components/TableHeadWrapper.jsx]: no, I do not think so.

Comment (haxxmaxx) [reply src/table/components/TableHeadWrapper.jsx]: I tried and it doesn't for some weird reason, might be some other listener by deafult on the mui component. anyway, just use `handleClickToFocusHead` here too. it should do the same thing

Comment (LiKang6688) [reply src/table/components/TableHeadWrapper.jsx]: [columnIndex, 0] the order is wrong, should be [0, columnIndex].  handleClickToFocusHead also calls when clicking label.

Comment (LiKang6688) [reply src/table/components/TableHeadWrapper.jsx]: just use handleClickToFocusHead here does not work.

**Pull request merged**